src/pages/setup/export-config.tsx

```diff
@@ -9,84 +9,104 @@ require("codemirror/lib/codemirror.css");
-                </li>
-                <li className="nav-item">
-                    <a className={`nav-link${exportLanguage==="yaml" ? " active show" : ""}`} onClick={this.handleTabClick} data-toggle="tab" href="#yaml">YAML</a>
-                </li>
-                </ul>
-                <CodeMirror
-                    value={configText}
-                    options={{
-                        readOnly: true,
-                        lineNumbers: true,
-                        mode: mode
-                    }}
-                />
-          </div>
-        );
+  public render() {
+    const { options } = this.props;
+    const { exportLanguage } = this.state;
+    let mode;
+    let configText;
+    if (exportLanguage === "yaml") {
+      mode = "yaml";
+      configText = jsYaml.dump(options);
+    } else {
+      mode = {
+        name: "javascript",
+        json: true,
+      };
+      configText = JSON.stringify(options, null, 2);
     }
+    return (
+      <div className="exportConfig">
+        <Download
+          file={`.unibeautifyrc.${exportLanguage}`}
+          content={configText}
+        >
+          <button className="btn btn-info" type="submit">
+            Download
+          </button>
+        </Download>
+        <CopyToClipboard text={configText}>
+          <button className="btn btn-outline-info" type="submit">
+            Copy
+          </button>
+        </CopyToClipboard>
+        <ul className="nav nav-tabs">
+          <li className="nav-item">
+            <a
+              className={`nav-link${
+                exportLanguage === "json" ? " active show" : ""
+              }`}
+              onClick={this.handleTabClick}
+              data-toggle="tab"
+              href="#json"
+            >
+              JSON
+            </a>
+          </li>
+          <li className="nav-item">
+            <a
+              className={`nav-link${
+                exportLanguage === "yaml" ? " active show" : ""
+              }`}
+              onClick={this.handleTabClick}
+              data-toggle="tab"
+              href="#yaml"
+            >
+              YAML
+            </a>
+          </li>
+        </ul>
+        <CodeMirror
+          value={configText}
+          options={{
+            readOnly: true,
+            lineNumbers: true,
+            mode: mode,
+          }}
+        />
+      </div>
+    );
+  }
 
-    private handleTabClick(event: React.MouseEvent<HTMLAnchorElement>) {
-        event.preventDefault();
-        const tab = event.target;
-        if (tab instanceof Element) {
-            let language = tab.getAttribute("href");
-            language = language ? language.slice(1, language.length) : "";            
-            this.setState({
-                exportLanguage: language
-            });
-        }        
-    }    
+  private handleTabClick(event: React.MouseEvent<HTMLAnchorElement>) {
```

Comment:
Instead of using

```
+    this.handleTabClick = this.handleTabClick.bind(this);
```

on line 22.

Do this:

```
-  private handleTabClick(event: React.MouseEvent<HTMLAnchorElement>) {
+  private handleTabClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
```

😃